src/libraries/System.Private.CoreLib/src/System/Globalization/NumberFormatInfo.cs

```diff
@@ -242,9 +263,19 @@ public string CurrencyDecimalSeparator
                 VerifyWritable();
                 ArgumentException.ThrowIfNullOrEmpty(value);
                 _currencyDecimalSeparator = value;
+                _currencyDecimalSeparatorUtf8 = null;
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal ReadOnlySpan<TChar> CurrencyDecimalSeparatorTChar<TChar>() where TChar : unmanaged, IBinaryInteger<TChar>
```

Comment:
Could be worth it to extract the duplicated type checks and casts to a helper method, e.g.:

```csharp
return GetOrInitTChar<TChar>(_currencyDecimalSeparator, ref _currencyDecimalSeparatorUtf8);
```

```csharp
[MethodImpl(MethodImplOptions.AggressiveInlining)]
private static ReadOnlySpan<TChar> GetOrInitTChar<TChar>(char[] text, ref byte[]? utf8)
    where TChar : unmanaged, IBinaryInteger<TChar>
{
    if (typeof(TChar) == typeof(char))
    {
        return MemoryMarshal.Cast<char, TChar>(text);
    }

    if (typeof(TChar) == typeof(byte))
    {
        return MemoryMarshal.Cast<byte, TChar>(utf8 ??= Encoding.UTF8.GetBytes(text));
    }

    throw new UnreachableException();
}

```

Reply:
We could.  It could possibly add some cost, though, as for example in some of the cases the char-based property is actually itself lazy initializing, so accessing the string to pass it in isn't just a field access.  Part of the issues this PR is addressing is also now todo with some constraints around inlining, and so I'm a little hesitant to put even more pressure on the inliners.  It's quite possible it would all "just work" nicely, but I'd like to avoid doing it here.  Thanks for the suggestion.